projects/components/src/datagrid/directives/component-renderer-outlet.directive.ts

```diff
@@ -32,24 +32,28 @@ export class ComponentRendererOutletDirective<R, T> {
     constructor(private viewContainerRef: ViewContainerRef, private cfr: ComponentFactoryResolver) {}
 
     @Input()
-    set vcdComponentRendererOutlet(renderer: { rendererSpec: ComponentRendererSpec<T>; context?: R }) {
+    set vcdComponentRendererOutlet(renderer: {
+        rendererSpec: ComponentRendererSpec<T>;
+        context?: R;
+        injector?: Injector;
+    }) {
         if (this.componentType !== renderer.rendererSpec.type) {
             // Cache the componentType to avoid redundant detaching and attaching of component to this host
             this.componentType = renderer.rendererSpec.type;
-            this.componentRef = this.attachRenderer();
+            this.componentRef = this.attachRenderer(renderer.injector);
         }
         this.assignValue(renderer.rendererSpec.config, renderer.context);
     }
 
     /**
      * Attaches the passed component type to the view of this directive host
      */
-    private attachRenderer(): ComponentRef<ComponentRenderer<T>> {
+    private attachRenderer(injector?: Injector): ComponentRef<ComponentRenderer<T>> {
         if (this.componentRef) {
             this.detachRenderer();
         }
         const componentFactory = this.cfr.resolveComponentFactory(this.componentType);
-        return this.viewContainerRef.createComponent(componentFactory);
+        return this.viewContainerRef.createComponent(componentFactory, 0, injector);
```

Comment:
If there is no injector defined, would this mess up the hierarchical injections this component might depend upon?

Reply:
In the new way of doing things, when we provide the LoadingListener because we are injecting it, we can assure that it is never undefined. We also know that all rows always provide LoadingListener. Does this sound correct?